#!/usr/bin/env python
PACKAGE = "tutorial_dynamic_reconfigure"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("planner_frequency",int_t,0,"global planning frequency(hz)",0,0,50)
gen.add("controller_frequency",int_t,0,"controller frequency(hz)",2,0,100)
gen.add("planner_name",str_t,0,"the name of planner")
gen.add("isDone",bool_t,0,"is done",True)

size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
                      gen.const("Medium",     int_t, 1, "A medium constant"),
                      gen.const("Large",      int_t, 2, "A large constant"),
                      gen.const("ExtraLarge", int_t, 3, "An extra large constant")], "An enum to set size")
                      
gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)
exit(gen.generate(PACKAGE,"tutorial_dynamic_reconfigure","tutorials"))   
